### ISSUE TITLE
Typescript error: connectors does not accept the null parameter.

### ISSUE THREAD

**Issue by Traveller23** [prevwong/craft.js#574]
**Describe the bug**
When I write the simplest example code in typescript, I get the following error:

![1](https://github.com/prevwong/craft.js/assets/117014547/550125db-63e8-4238-9224-80d0686a9f1e)
```typescript
import {useNode} from "@craftjs/core";
import Button from '@mui/material/Button';
export const MyComponent = () => {
    const {connectors: {connect, drag}} = useNode();
    return <Button ref={ref => connect(drag(ref))}/>;
}
```

![2](https://github.com/prevwong/craft.js/assets/117014547/61c80eb0-85df-4f1a-82ce-93166abe11b2)

> TS2345: Argument of type  `HTMLButtonElement | null`  is not assignable to parameter of type
> `HTMLElement | ReactElement<any, string | JSXElementConstructor<any>>`
> Type  `null`  is not assignable to type
> `HTMLElement | ReactElement<any, string | JSXElementConstructor<any>>`

I think the reason for this error is that the connector methods do not accept null as argument, but the `ref` can actually be null:

![3](https://github.com/prevwong/craft.js/assets/117014547/a897430c-fa35-428e-a39e-53fd7e877913)

> Allows getting a ref to the component instance. Once the component unmounts, React will set ref.current to null (or call the ref with null if you passed a callback ref).
> See also: https://react.dev/learn/referencing-values-with-refs#refs-and-the-dom

*file: @types/react/index.d.ts*

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| craft.js         | 0.2.2
| React            | 18.2.0
| TypeScript       | 5.2.2
| Browser          | -
| npm/Yarn         | -
| Operating System | -

**Comment by Traveller23** [comment]
Indeed, I do know how to get around this problem, i.e. write `drag(ref!)` instead.
But I'm concerned that craft.js doesn't do anything special with null values, which could lead to problems that are currently invisible but could be far-reaching.

**Comment by th-kahan-sonani** [comment]
try this `(ref) => ref && drag(ref)`
